utils/notifications.ts

```diff
@@ -40,161 +46,98 @@ export const lastNotifSubscribeByAccount: {
-  //     );
-  //     topicsToUpdateForPeriod[userGroupInviteTopic] = {
-  //       status: "PUSH",
-  //     };
-  //     dataToHash.push.push(userGroupInviteTopic);
-  //     dataToHash.push.sort();
-  //     dataToHash.muted.sort();
-  //     const stringToHash = `${dataToHash.period}-push-${dataToHash.push.join(
-  //       ","
-  //     )}-muted-${dataToHash.muted.join(",")}`;
-  //     lastNotifSubscribeByAccount[account] =
-  //       lastNotifSubscribeByAccount[account] || {};
-  //     const lastStringToHash = lastNotifSubscribeByAccount[account]?.stringToHash;
-  //     if (stringToHash === lastStringToHash) {
-  //       delete subscribingByAccount[account];
-  //       return;
-  //     }
-  //     const hash = (
-  //       await createHash(Buffer.from(stringToHash), "sha256")
-  //     ).toString("hex");
-  //     if (!lastNotifSubscribeByAccount[account]?.serverHash) {
-  //       lastNotifSubscribeByAccount[account].serverHash =
-  //         await getLastNotificationsSubscribeHash(account, nativePushToken);
-  //     }
-  //     if (
-  //       lastNotifSubscribeByAccount[account] &&
-  //       lastNotifSubscribeByAccount[account]?.serverHash === hash
-  //     ) {
-  //       lastNotifSubscribeByAccount[account].stringToHash = stringToHash;
-  //       // We're already up to date!
-  //       delete subscribingByAccount[account];
-  //       return;
-  //     }
-  //     logger.info(
-  //       `[Notifications] Subscribing to ${
-  //         Object.keys(topicsToUpdateForPeriod).length
-  //       } topic for ${account}`
-  //     );
-  //     lastNotifSubscribeByAccount[account].serverHash =
-  //       await saveNotificationsSubscribe(
-  //         account,
-  //         nativePushToken,
-  //         nativeTokenQuery.type,
-  //         nativeTokenQuery.type === "android" ? "converse-notifications" : null,
-  //         topicsToUpdateForPeriod
-  //       );
-  //     // Also saved the hashed string so we don't hash it again with no reason
-  //     lastNotifSubscribeByAccount[account].stringToHash = stringToHash;
-  //     subscribedOnceByAccount[account] = true;
-  //   } catch (e) {
-  //     logger.error(e, { context: "Error while subscribing to notifications" });
-  //   }
-  //   delete subscribingByAccount[account];
+type SubscribeToNotificationsParams = {
+  conversations: ConversationWithCodecsType[];
+  account: string;
 };
 
-// Don't call twice in 1 sec
-export const subscribeToNotifications = debounce(
-  _subscribeToNotifications,
-  1000
-);
+export const subscribeToNotifications = async ({
+  conversations,
+  account,
+}: SubscribeToNotificationsParams): Promise<void> => {
+  try {
+    logger.info(
+      "[subscribeToNotifications] start",
+      account,
+      conversations.length
+    );
+    const thirtyDayPeriodsSinceEpoch = Math.floor(
+      Date.now() / 1000 / 60 / 60 / 24 / 30
+    );
+    const client = (await getXmtpClient(account)) as ConverseXmtpClientType;
```

Comment:
I see we do A LOT of those `const client = (await getXmtpClient(account)) as ConverseXmtpClientType;` could we make it simpler and at least not need the `as ... ` just curious

Reply:
Yeah I agree we don't need it, it's there because of old web stuff, so we should probably remove it